var/spack/repos/builtin/packages/genie/package.py

```diff
@@ -0,0 +1,265 @@
+        sha256="7ca169a8d9eda7267d28b76b2f3110552852f8eeae263a03cd5139caacebb4ea",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_12_8.tar.gz",
+    )
+    version(
+        "2.12.6",
+        sha256="3b450c609875459798ec98e12cf671cc971cbb13345af6d75bd6278d422f3309",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_12_6.tar.gz",
+    )
+    version(
+        "2.12.4",
+        sha256="19a4a1633b0847a9f16a44e0c74b9c224ca3bb93975aecf108603c22e807517b",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_12_4.tar.gz",
+    )
+    version(
+        "2.12.2",
+        sha256="cbdc45a739878940dadcaaed575b5cad6b5e7035f29605045b1ca557e6faa6d1",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_12_2.tar.gz",
+    )
+    version(
+        "2.12.0",
+        sha256="d2b01c80f38d269cb0296b3f2932798ef3f1d51bd130e81274fbfeeb381fac6b",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_12_0.tar.gz",
+    )
+    version(
+        "2.11.2",
+        sha256="0f4c25d8ceb7513553671643c9cdac5aa98c40fc8594a5ecb25c077c6b36166e",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_11_2.tar.gz",
+    )
+    version(
+        "2.11.0",
+        sha256="1ebe0eb65d797595413632f1cec1cb2621cb8e8d0384a2843799724a79b1d80c",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_11_0.tar.gz",
+    )
+    version(
+        "2.10.10",
+        sha256="1dfaadcf1bbaf6e164b612f410c4399301e63497ad6a4891706b1787ac11a7a1",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_10_10.tar.gz",
+    )
+    version(
+        "2.10.8",
+        sha256="4f6f5af2062e7c505b76e70547ac2ae304a9790c3e9b9592818d8aebeebc8398",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_10_8.tar.gz",
+    )
+    version(
+        "2.10.6",
+        sha256="d00b4288c886f81459fb2967e539f30315d4385f82d1d3f4330298d313f9a176",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_10_6.tar.gz",
+    )
+    version(
+        "2.10.4",
+        sha256="df909bf7e1a789ca01794995687da2af803769f0823273a4a3a31678d6d5b0f1",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_10_4.tar.gz",
+    )
+    version(
+        "2.10.2",
+        sha256="6abe4e0cdb5e8f5beddf0ccdbebc94c175a9f72592b1cbbffe01b88ee3972bf9",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_10_2.tar.gz",
+    )
+    version(
+        "2.10.0",
+        sha256="17bda900c996b6f4f10a7f6a3be94e56c3b8dcdeb2ef8865ca7f20c5fe725291",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_10_0.tar.gz",
+    )
+    version(
+        "2.9.0",
+        sha256="8229beb73f65f5af86a77bf141acfbe4a8b68cba9d797aae083a929906f6f2a2",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_9_0.tar.gz",
+    )
+    version(
+        "2.8.6",
+        sha256="310dc8e0d17a65e6b9773e398250703a3a6f94ceafe94f599ae0f7b3fecf7e6c",
+        url="https://github.com/GENIE-MC/Generator/archive/R-2_8_6.tar.gz",
+    )
```

Comment:
The `version` lines get an exception for the Spack linter tools, so it makes the path more readable to put the version/checksum on the same line.  Since the version numbers are mostly formulaic (based on the tags) you should delete the custom `url` and replace with the special method:
```python
    def url_for_version(self, version):
        url = "https://github.com/GENIE-MC/Generator/archive/R-{0}.tar.gz"
        if version >= Version(3):
            return url.format("{0}_{1:02d}_{2:02d}".format(*version))
        else:
            return url.format(version.underscored)
```

Reply:
Thanks. Implemented in 240eb22 .